tests/pages/auth/login.spec.js

```diff
@@ -0,0 +1,148 @@
+    i18n: { changeLanguage: jest.fn() }, // Mock language change function
+  }),
+}))
+jest.mock('firebase/auth', () => ({
+  getAuth: jest.fn(), // Mock getAuth function
+  sendPasswordResetEmail: jest.fn(), // Mock sendPasswordResetEmail function
+}))
+
+describe('Testing the signIn page', () => {
+  beforeAll(() => {
+    // Mock the useAuth hook to provide mock functions for signup and login methods
+    useAuth.mockReturnValue({
+      signup: jest.fn(),
+      loginGoogle: jest.fn(),
+      loginGithub: jest.fn(),
+    })
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks() // Clear mocks after each test to avoid test interference
+  })
+
+  it('Renders the login form', () => {
+    render(<AuthPage />) // Render the AuthPage component
+
+    // Check if the required elements are present in the document
+    expect(screen.getByText('buttons.register_now')).toBeInTheDocument() // Register now button
+    expect(screen.getByLabelText('E-mail')).toBeInTheDocument() // Email input field
+    expect(screen.getByLabelText('form.password')).toBeInTheDocument() // Password input field
+    expect(screen.getByText('buttons.log_in')).toBeInTheDocument() // Log in button
+  })
+
+  it('Calls the login function with correct data', async () => {
+    const mockLogin = jest.fn() // Create a mock login function
+    useAuth.mockReturnValue({ login: mockLogin }) // Override useAuth to use the mock function
+
+    const email = faker.internet.email() // Generate a fake email
+    const password = faker.internet.password() // Generate a fake password
+
+    render(<AuthPage />) // Render the AuthPage component
+
+    // Fill in the login form
+    fireEvent.change(screen.getByLabelText('E-mail'), { target: { value: email } })
+    fireEvent.change(screen.getByLabelText('form.password'), { target: { value: password } })
+    fireEvent.click(screen.getByText('buttons.log_in')) // Click the login button
+
+    // Wait for the login function to be called with the correct parameters
+    await waitFor(() => {
+      expect(mockLogin).toHaveBeenCalledWith({ email, password })
+    })
+  })
+
+  it('Toggles between login and registration forms', () => {
+    render(<AuthPage />) // Render the AuthPage component
+
+    // Check if the register now button is present and click it
+    expect(screen.getByText('buttons.register_now')).toBeInTheDocument()
+    fireEvent.click(screen.getByText('buttons.register_now'))
+
+    // Verify that the sign-in button is now present
+    expect(screen.getByText('buttons.sign_in')).toBeInTheDocument()
+
+    // Click the sign-in button to toggle back
+    fireEvent.click(screen.getByText('buttons.sign_in'))
+    expect(screen.getByText('buttons.register_now')).toBeInTheDocument() // Check that register now button is visible again
+  })
+
+  // BUG: In pages/auth/index.js I can't find the button buttons.recover_password and nowhere in the codebase but in production this text appears
```

Comment:
This is at `pages/auth/index.js` at line 146: `handleResetPassword()`.

I think the recover_password was left over at locales and the correct button is being used as `forgot_password`